farmdata2/farmdata2_modules/resources/BannerComponent.js

```diff
@@ -0,0 +1,113 @@
+/**
+ * A Vue component for a message banner that is displayed when a specific event (success/error/etc.) occurs.
+ * 
+ * <h3 class="subsection-title">data-cy attributes</h3>
+ * <table>
+ * <thead><tr><th>Value</th>        <th>Description</th></tr></thead>
+ * <tbody>
+ * <tr><td>banner-handler</td>   <td>div containing bootstrap alert banner</td></tr>
+ * <tr><td>banner-close</td>   <td>'X' button that closes the alert banner</td></tr>
+ * <tr><td>banner-message</td>   <td>The message displayed inside the alert banner</td></tr>
+ * </tbody>
+ * </table>
+ * 
+ * <div>
+ *  @vue-prop {Object} [message={"msg": "Hello, I am a banner alert.", "class": "alert alert-warning"}] - Contains the message and class for the alert banner
+ *  @vue-prop {Boolean} [visible=null] - Set to true to show the alert banner. False makes the banner invisible.
+ *  @vue-prop {Boolean} [timeout=null] - Sets whether or not the banner dismisses itself which disables the 'x' to close. Set to false by default.
+ * </div>
+ * 
+ * <div>
+ *  @vue-event visibility-update - Updates the visibility in the parent page when the child component has internal visibility updates. This means each page needs to implement a function that catches this emit and matches their visibility with the child's internal visibility. 
+ * </div>

```

Comment:
This component should have a code example that illustrates how a page would implement the handler.  See some of the functions in the `FarmOSAPI` documentation for examples of how to add code examples.  You don't necessarily need a complete example... but something a reader could easily adapt.  I.e. show the component with its prop bound and the event handler assigned, the data property and the event handler.

Also, when you say "Updates the visibility in the parent page..." that seems to suggest that this event does that.  However, the event doesn't do that.  It rather indicates that the banner has been hidden (I think... does this fire when the banner is shown too?).  So maybe a renaming here would make sense?   Call the event `banner-hidden` and then say something like "event indicating that the banner has been hidden either by the timeout or by the user clicking the close X.  The page should use this event to..."

Reply:
You're correct, This emitted event only emits when the banner is hidden, so it would be better to call the event `banner-hidden` since it only fires when the banner is hidden. I don't believe there needs to be any update when the visibility is updated to true on the parent side since that'll go down into the child component. Then the hiding of the banner is handled by the child which communicates with the parent to inform it that the banner component is now hidden and its visibility value should now be false. 

I'll look into the `FarmOSAPI` and make sure to provide an example.